pkg/internal/infrastructure/template.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package infrastructure
+
+import (
+	"bytes"
+	_ "embed"
+	"fmt"
+	"text/template"
+
+	"github.com/Masterminds/sprig"
+	"github.com/pkg/errors"
+)
+
+var (
+	//go:embed templates/main.tpl.tf
+	mainFile string
+	//go:embed templates/terraform.tfvars
+	terraformTFVars []byte
+	//go:embed templates/variables.tf
+	variablesTF string
```

Comment:
Why did you model them as templates compared to bare string constants?

Reply:
At the very least I think that the `variables.tf` is a better match for a file that we can edit, compared to an inline raw string.

For a more nuanced reason, by doing the swap away from helm we "lose" access to a possible, albeit rare workflow: to render the helm chart directly and use terraform from the CLI for quick validations. I am not sure if we will continue using that (sure we can have a small "dumb" app to do template rendering if we want), but for the time being I like that the file structure is that of a "normal" terraform project and keep the possibility open.

Plus `variables.tf` IDE completion :)